### PR TITLE
Feature/coinify

### PR DESCRIPTION
**Pull request by prwelber**
This PR begins the main implementation of Coinify Buy.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/components/StepIndicator/index.js [hunk 1]
```diff
@@ -49,8 +50,8 @@ const Steps = styled.div`
 
 const Step = styled.span`
   font-size: 14px;
-  min-width: 70px;
-  max-width: 70px;
+  min-width: ${props => props.coinify ? '135px' : '70px'};
+  max-width: ${props => props.coinify ? '135px' : '70px'};
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/components/StepIndicator/index.js]: This component shouldn't know anything about who is using it. If we want to give it a width property that's fine.

File: packages/blockchain-wallet-v4-frontend/src/components/StepIndicator/index.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ const Wrapper = styled.div`
   flex-direction: row;
   align-items: center;
   width: 100%;
+  justify-content: ${props => props.coinify ? 'flex-end' : ''};
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/components/StepIndicator/index.js]: Would rather we do something more generic here.

File: packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js [hunk 3]
```diff
@@ -11,8 +14,10 @@ export default ({ coreSagas }) => {
       yield call(coreSagas.data.coinify.signup)
       const profile = yield select(selectors.core.data.coinify.getProfile)
       if (!profile.error) {
-        yield put(A.coinifyNextStep('order'))
-        yield put(actions.alerts.displaySuccess('Account successfully created!'))
+        // yield put(actions.modals.closeAllModals())
+        yield call(coreSagas.data.coinify.triggerKYC)
+        yield put(A.coinifyNextStep('isx'))
+        // yield put(actions.alerts.displaySuccess('Account successfully created!'))
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js]: Comments

File: packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js [hunk 4]
```diff
@@ -2,6 +2,9 @@ import { put, call, select } from 'redux-saga/effects'
 import * as A from './actions'
 import * as actions from '../../actions'
 import * as selectors from '../../selectors.js'
+// import { formValueSelector } from 'redux-form'
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js]: Comment

File: packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/CoinifyCheckout/Content/QuoteInputTemplate.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    padding-right: 5px;
+  }
+`
+
+const getLimitsError = (errorType, limits, symbol, setMin) => {
+  if (errorType === 'below_min') return `Your limit of ${symbol}${limits.max} is below the minimum allowed amount.`
+  if (errorType === 'over_max') return `Enter an amount under your ${symbol}${limits.max.toLocaleString()} limit`
+  // if (errorType === 'under_min') return `Enter an amount above the ${symbol}${limits.min.toLocaleString()} minimum`
+  if (errorType === 'under_min') return <FormattedMessage id='buy.quote_input.under_min' defaultMessage='Enter an amount above the {setMin} minimum' values={{ setMin: <a onClick={() => setMin(limits.min)}>{symbol}{limits.min.toLocaleString()}</a> }} />
+  // if (errorType === 'effective_max') return `Enter an amount less than your balance minus the priority fee (${limits.effectiveMax / 1e8} BTC)`
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/CoinifyCheckout/Content/QuoteInputTemplate.js]: Comments

File: packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/CoinifyCheckout/Content/Sell/selectors.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import { formValueSelector } from 'redux-form'
+import { lift, path } from 'ramda'
+// import settings from 'config'
+import { selectors } from 'data'
+
+export const getProfileData = (state) => {
+  const profile = selectors.core.data.coinify.getProfile(state)
+  // const nextAddress = selectors.core.common.bitcoin.getNextAvailableReceiveAddress(settings.NETWORK_BITCOIN, 0, state)
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/CoinifyCheckout/Content/Sell/selectors.js]: Why is this commented, don't we need the next address?

File: packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/CoinifyCheckout/Content/Sell/template.success.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+      <Stepper initialStep={0}>
+        <StepView step={0}>
+          <CheckoutWrapper>
+            <OrderCheckout
+              quoteR={sellQuoteR}
+              rateQuoteR={rateQuoteR}
+              onFetchQuote={fetchSellQuote}
+              limits={limits.sell}
+              type={'sell'}
+              reason={'has_remaining'} // placeholder for now - coinify does not require a reason
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/BuySell/CoinifyCheckout/Content/Sell/template.success.js]: Here are a few reasons that we had to handle at at least one point:

https://github.com/blockchain/My-Wallet-V3-Frontend/blob/master/assets/js/services/coinify.service.js#L62-L66

File: packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js (hunk 3 above)
Comment (prwelber) [reply packages/blockchain-wallet-v4-frontend/src/data/modules/coinify/sagas.js]: Wasn't quite finished with this one yet.

**Comment by prwelber** [comment]
@plondon all comments with a 👍  have been resolved. I'll look into the other ones in the next Buy PR. Thanks!

**Comment by plondon** [comment]
Thanks @prwelber !

**Pull request merged**